Zend/zend_builtin_functions.c

```diff
@@ -2447,12 +2447,15 @@ ZEND_FUNCTION(debug_print_backtrace)
 
 		if (call->func) {
 			func = call->func;
-			function_name = (func->common.scope &&
-			                 func->common.scope->trait_aliases) ?
-				ZSTR_VAL(zend_resolve_method_name(
-					(object ? object->ce : func->common.scope), func)) :
-				(func->common.function_name ?
-					ZSTR_VAL(func->common.function_name) : NULL);
+			zend_string *zend_function_name;
+			if (func->common.scope && func->common.scope->trait_aliases) {
+				zend_function_name = zend_resolve_method_name(object ? object->ce : func->common.scope, func);
+			} else {
+				zend_function_name = func->common.function_name;
+			}
+			if (zend_function_name != NULL) {
+				function_name = ZSTR_VAL(zend_function_name);
+			}
```

Comment:
I'd add in an
```
else {
    function_name = NULL;
}
```
here, to ensure that the function name is `NULL` if it does not exist (since currently, it could be any garbage). Without it, I get the following output (in debug mode):
```
#0  c1::() called at [/Users/tpunt/prog/php/tpuntphp/a.php:20]
#1  c1::go() called at [/Users/tpunt/prog/php/tpuntphp/a.php:24]
```